internal/cli/atlas/networking/peering/create/aws.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package create
+
+import (
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/output"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type AwsOpts struct {
+	cli.GlobalOpts
+	region              string
+	routeTableCidrBlock string
+	accountID           string
+	containerID         string
+	vpcID               string
+	atlasCIDRBlock      string
+	store               store.PeeringConnectionCreator
+}
+
+func (opts *AwsOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *AwsOpts) Run() error {
+	container, err := opts.containerExists()
+	if err != nil {
+		return err
+	}
+
+	if container == nil {
+		var err2 error
+		container, err2 = opts.store.CreateContainer(opts.ConfigProjectID(), opts.newContainer())
+		if err2 != nil {
+			return err2
+		}
+	}
+	r, err := opts.store.CreatePeeringConnection(opts.ConfigProjectID(), opts.newPeer(container.ID))
+	if err != nil {
+		return err
+	}
+	return output.Print(config.Default(), createTemplate, r)
+}
+
+func (opts *AwsOpts) containerExists() (*atlas.Container, error) {
+	r, err := opts.store.AllContainers(opts.ConfigProjectID(), nil)
+	if err != nil {
+		return nil, err
+	}
+	for i := range r {
+		region := strings.ToUpper(opts.region)
+		region = strings.ReplaceAll(region, "-", "_")
```

Comment:
I am not using the same arguments for `ReplaceAll`. Indeed,
Container: Region = `EU_WEST_1`
Peer: Region = `eu-west-1`

Those two lines allow us to have only one flag `region` instead of creating also `AccepterRegionName`.
Given that `ReplaceAll` is called in two different place with different arguments I wouldn’t create a function for it.

Reply:
lines 73-74
```go
region := strings.ToUpper(opts.region)
region = strings.ReplaceAll(region, "-", "_")
```
lines  83-84
```go
region := strings.ToUpper(opts.region)
region = strings.ReplaceAll(region, "-", "_")
```

They look the same to me

Reply:
ops you’re right 😅